### DISCUSSION TITLE
OpenTelemetry Tracer not injecting in Q2.16

### DISCUSSION THREAD

**Discussion by vkrizan** [quarkusio/quarkus#36721] [Q&A]
Hello,

We're migrating from OpenTracing to OpenTelementry, however we are hitting an issue with CDI for `io.opentelemetry.api.trace.Tracer`:
```
Unsatisfied dependency for type io.opentelemetry.api.trace.Tracer and qualifiers
```
(https://github.com/RedHatInsights/policies-ui-backend/pull/588#issuecomment-1779378921)

Base on the [documentation|https://quarkus.io/version/2.16/guides/opentelemetry#available-opentelemetry-cdi-injections] this should be supported in Quarkus 2.16.
What are we missing?

(Our changes can be found at https://github.com/RedHatInsights/policies-ui-backend/pull/588/files)

Thank you for any help!

[a comment by an automated account is not shown]

**Comment by brunobat** [comment]
The bean producer is there: https://github.com/quarkusio/quarkus/blob/d41313fb1af4c3c43cbc709c28089732a8980977/extensions/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/cdi/TracerProducer.java#L34

**Comment by vkrizan** [comment]
@brunobat so the producer should be auto-registered by including quarkus-opentelemetry, right? Do you happen to know why it is not injecting?

**Comment by vkrizan** [comment]
I'm tracing arc processor and I do not see that producer being registered. I'll double check package versions, but I feel like that it is not being picked up for some reason.

**Comment by vkrizan** [comment]
I think the problem is with opentracing being disabled with `quarkus.opentelemetry.enabled=false`.
